src/app/goals/goals-list/goals-list.component.ts

```diff
@@ -42,6 +46,19 @@ export class GoalsListComponent implements OnInit, OnDestroy {
       this.goals = goals;
       this.selectTheFirstGoal();
     });
+
+    this.filter.valueChanges
+      .pipe(
+        distinctUntilChanged(),
+        debounceTime(200),
+        switchMap(value =>
+          this.goalsQuery.selectAll({
```

Comment:
Could you extract it to separate private method like `getFilteredGoals(filter: String)` and just call it here.